docs/openstack/management/vm-management.md

```diff
@@ -0,0 +1,300 @@
+address the problem, the virtual machine console offers valuable diagnostic
+information on the underlying cause.
+
+Alternatively, utilizing OpenStack's rescue functions involves booting the
+virtual machine using the original image, with the system disk provided as a
+secondary disk. This allows manipulation of the disk, such as using `fsck` to
+address filesystem issues or mounting and editing the configuration.
+
+!!! warning "Important Note"
+    We cannot rescue a volume-backed instance that means **ONLY** instance running
+    using [Ephemeral disk](../persistent-storage/volumes.md#ephemeral-disk) can
+    be rescued. Also, this procedure has not been tested for Windows virtual machines.
+
+VMs can be rescued using either the OpenStack dashboard by clicking
+*Action -> Rescue Instance* or via the openstack client using
+`openstack server rescue ...` command.
+
+If however, the virtual machine is no longer required and no data on the
+associated system or ephemeral disk needs to be preserved, the following command
+can be run:
+
+    openstack server rescue <INSTANCE_NAME_OR_ID>
+
+or, using Horizon dashboard:
+
+Navigate to Project -> Compute -> Instances.
+
+Select an instance.
+
+Click *Action -> Rescue Instance*.
+
+!!! warning "When to use Rescue Instance"
+    The rescue mode is only for emergency purpose, for example in case of a
+    system or access failure. This will shut down your instance and mount the
+    root disk to a temporary server. Then, you will be able to connect to this
+    server, repair the system configuration or recover your data. You may
+    optionally select an image and set a password on the rescue instance server.
+
+![Rescue Instance Popup](images/rescue_instance_popup.png)
+
+### Troubleshoot the disk
+
+This will reboot the virtual machine and you can then log in using the key pair
+previously defined. You will see two disks, `/dev/vda` which is the new system disk
+and `/dev/vdb` which is the old one to be repaired.
+
+    ubuntu@my-vm:~$ lsblk
+    NAME    MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
+    loop0     7:0    0   62M  1 loop /snap/core20/1587
+    loop1     7:1    0 79.9M  1 loop /snap/lxd/22923
+    loop2     7:2    0   47M  1 loop /snap/snapd/16292
+    vda     252:0    0  2.2G  0 disk
+    ├─vda1  252:1    0  2.1G  0 part /
+    ├─vda14 252:14   0    4M  0 part
+    └─vda15 252:15   0  106M  0 part /boot/efi
+    vdb     252:16   0   20G  0 disk
+    ├─vdb1  252:17   0 19.9G  0 part
+    ├─vdb14 252:30   0    4M  0 part
+    └─vdb15 252:31   0  106M  0 part
+
+The old one can be mounted and configuration files edited or `fsck`'d.
+
+    # lsblk
+    # cat /proc/diskstats
+    # mkdir /tmp/rescue
+    # mount /dev/vdb1 /tmp/rescue
+
+### Unrescue the VM
+
+On completion, the VM can be returned to active state with
+`openstack server unrescue ...` openstack client command, and rebooted.
+
+    openstack server unrescue <INSTANCE_NAME_OR_ID>
```

Comment:
is it worth mentioning around here that if they use instance name with spaces they need to encapsulate it?

Reply:
same with any place we suggest the use of instance name.

Reply:
I added this in line - 124